### PR TITLE
SimpleQ.S.B and QueryStringQ.S.B tests should avoid `now` in query

### LINKED ISSUE #43112: [CI] SimpleQueryStringBuilderTests.testToQuery failure
`SimpleQueryStringBuilderTests.testToQuery` failed in a 7.2 build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.2+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=zulu8,nodes=immutable&&linux&&docker/13/console

The error is:

```
org.elasticsearch.index.query.SimpleQueryStringBuilderTests > testToQuery FAILED
    java.lang.AssertionError: query was marked as cacheable in the context but this test indicates it should not be cacheable: {
      "simple_query_string" : {
        "query" : "NoWt",
        "flags" : -1,
        "default_operator" : "or",
        "analyze_wildcard" : false,
        "quote_field_suffix" : "",
        "auto_generate_synonyms_phrase_query" : true,
        "fuzzy_prefix_length" : 4,
        "fuzzy_max_expansions" : 5,
        "fuzzy_transpositions" : true,
        "boost" : 1.0
      }
    }
```

This reliably reproduces on the 7.2 branch using this command:

```
./gradlew :server:test --tests "org.elasticsearch.index.query.SimpleQueryStringBuilderTests.testToQuery"   -Dtests.seed=9492DAEC1EE2893   -Dtests.security.manager=true   -Dtests.locale=lv-LV   -Dtests.timezone=Europe/Dublin   -Dcompiler.java=12   -Druntime.java=8
```

So it seems like a seed-specific failure.

### PR DESCRIPTION
**Pull request by cbuescher**
Currently the randomization of the q.b. in these tests can create query strings
that can cause caching to be disabled for this query if we query all fields and
there is a date field present. This is pretty much an anomaly that we shouldn't
generally test for in the "testToQuery" tests where cache policies are checked.

This change makes sure we don't create offending query strings so the cache
checks never hit these cases and adds a special test method to check this edge
case.

Closes #43112

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java [hunk 1]
```diff
@@ -786,4 +784,34 @@ private void assertQueryWithAllFieldsWildcard(Query query) {
         assertThat(disjunctionMaxQuery.getDisjuncts(), hasItems(new TermQuery(new Term(STRING_FIELD_NAME, "hello")),
             new TermQuery(new Term(STRING_FIELD_NAME_2, "hello"))));
     }
+
+    /**
+     * Query terms that start with "now" can trigger a query to not be cacheable if it hits all field
+     * and there is a date field amongst them. This test checks the search context cacheable flag is
+     * updated accordingly
+     * Note: this cannot happen when directly hitting the date field, dates are not parsed leniently then
+     * and we typically get a parse exception.
```
Comment (cbuescher) [inline server/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java]: I’m not 100% sure we need this entire test, we should only be hitting the non-caching case when we query “all” fields and are in a lenient mode.

File: server/src/test/java/org/elasticsearch/index/query/QueryStringQueryBuilderTests.java [hunk 2]
```diff
@@ -84,6 +85,7 @@
 import static org.hamcrest.Matchers.instanceOf;
 
 public class QueryStringQueryBuilderTests extends FullTextQueryTestCase<QueryStringQueryBuilder> {
```
Comment (jimczi) [inline server/src/test/java/org/elasticsearch/index/query/QueryStringQueryBuilderTests.java]: Can we remove the `FullTextQueryTestCase` entirely if we ensure that `now` cannot be part of the query string.
You'll need to add dedicated tests in MatchQueryBuilderTests and MultiMatchQueryBuilderTests but we need to do that anyway so... There is a check in `FullTextQueryTestCase` that is related to fuzziness but it is wrong and should be removed anyway (fuzziness doesn't disable caching).

**Review by jimczi — COMMENTED** [review]
Thanks @cbuescher , I left a comment

**Comment by cbuescher** [comment]
@jimczi thanks, FullTextQueryTestCase is gone, "now" shoudln't be used as query text in the Simple- and QueryStringQB and also added dedicated date field cache test to MultiMatch- and MatchQB.

**Review by jimczi — APPROVED** [review]
LGTM

**Comment by cbuescher** [comment]
@jimczi thanks, will also backport this to active 7.x branches

**Pull request merged**